src/App.tsx

```diff
@@ -1,24 +1,208 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, {
+  FormEvent,
+  useCallback,
+  useEffect,
+  useMemo,
+  useState,
+} from 'react';
 import { UserWarning } from './UserWarning';
+import { TodoAppHeader } from './components/TodoAppHeader/TodoAppHeader';
+import { TodoAppContent } from './components/TodoAppContent/TodoAppContent';
+import { TodoAppFooter } from './components/TodoAppFooter/TodoAppFooter';
+import { Notifications } from './components/Notifications/Notifications';
+import { client } from './utils/fetchClient';
+import { Todo } from './types/Todo';
+import { FilterType } from './types/FilterType';
+import { ErrorType } from './types/ErrorType';
+import { TodoPostData } from './types/TodoPostData';
 
-const USER_ID = 0;
+const USER_ID = 10308;
+
+const getTodos = () => client.get<Todo[]>(`/todos?userId=${USER_ID}`);
+const postTodo = (data: TodoPostData) => client.post<Todo>(`/todos?userId=${USER_ID}`, data);
+const deleteTodo = (todoId: number) => client.delete(`/todos/${todoId}`);
+
+const prepareTodos = (todoList: Todo[], filterType: FilterType) => (
+  todoList.filter(todo => {
+    switch (filterType) {
+      case FilterType.ACTIVE:
+        return !todo.completed;
+
+      case FilterType.COMPLETED:
+        return todo.completed;
+
+      default:
+        return true;
+    }
+  })
+);
+
+const getActiveTodosCount = (todoList: Todo[]) => (
+  todoList.filter(todo => !todo.completed).length
+);
 
 export const App: React.FC = () => {
+  const [todoList, setTodoList] = useState<Todo[] | null>(null);
+  const [filterType, setFilterType] = useState(FilterType.ALL);
+  const [errorType, setErrorType] = useState(ErrorType.NONE);
+  const [isErrorShown, setIsErrorShown] = useState(false);
+  const [todoInputValue, setTodoInputValue] = useState('');
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [isAddDisabled, setIsAddDisabled] = useState(false);
+  const [deletingId, setDeletingId] = useState<number | null>(null);
+
+  const getTodoList = useCallback(async () => {
+    const todos = await getTodos();
+
+    setTodoList(todos);
+  }, []);
+
+  const preparedTodos = useMemo(() => (
+    prepareTodos(todoList || [], filterType)
+  ), [todoList, filterType]);
+
+  const activeTodosCount = useMemo(() => (
+    getActiveTodosCount(todoList || [])
+  ), [todoList]);
+
+  const areCompletedTodos = todoList
+    ? activeTodosCount < todoList.length
+    : false;
+
+  const handleFilterChange = (newFilterType: FilterType) => {
+    setFilterType(newFilterType);
+  };
+
+  const handleTodoInputChange = (value: string) => {
+    setIsErrorShown(false);
+
+    setTodoInputValue(value);
+  };
+
+  const executePostTodo = () => {
```

Comment:
wrap with `useCallback`
Also, choose one of the ways
1. `async / await`
2. `.then / .catch / .finally`
![image](https://github.com/mate-academy/react_todo-app-add-and-delete/assets/91411692/67d7dccc-9ab2-468e-8e15-577e597bfa77)